TypeScript/test/jest/gilded-rose.spec.ts

```diff
@@ -1,9 +1,259 @@
-import { Item, GildedRose } from '@/gilded-rose';
+import { Item, GildedRose, ITEM_UPDATE_STRATEGIES } from "@/gilded-rose";
 
-describe('Gilded Rose', () => {
-  it('should foo', () => {
-    const gildedRose = new GildedRose([new Item('foo', 0, 0)]);
-    const items = gildedRose.updateQuality();
-    expect(items[0].name).toBe('fixme');
+describe("Gilded Rose", () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe("Regular items", () => {
```

Comment:
Most of these tests can be collapsed into a few parametrised tests using `it.each()`. However, it felt like over-engineering and would harm readability and simplicity.